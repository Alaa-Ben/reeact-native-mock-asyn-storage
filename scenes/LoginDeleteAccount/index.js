import React, { Component } from 'react'
import {
  Text,
  View,
  Image,
  FlatList,
  StyleSheet,
  Linking
} from 'react-native'
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen'
import { Button } from 'react-native-elements'
import Icon from 'react-native-vector-icons/FontAwesome'
import { Col, Row, Grid } from 'react-native-easy-grid'
import AsyncStorage from '@react-native-community/async-storage'
import SplashScreen from 'react-native-splash-screen'

export default class LoginDeleteAccount extends Component {

  constructor(props) {
    super(props)
    this.state = {
      emails: []
    }
    this.renderItem = this.renderItem.bind(this)
  }

  async retrieveEmails() {
    try {
      const emails_array = await AsyncStorage.getItem('emails')
      if (emails_array !== null) this.setState({
        emails: JSON.parse(emails_array)
      })
    } catch (error) {
      console.log(error)
    }
  }

  redirectToNewAccount() {
    const { navigate } = this.props.navigation
    navigate('LoginNewAccount')
  }

  deleteAccount(email) {
    let emails = this.state.emails
    for (let index in emails) {
      if (emails[index] === email) {
        emails.splice(index, 1)
        this.setState({
          emails: emails
        })
        AsyncStorage.setItem('emails', JSON.stringify(this.state.emails))
        if (this.state.emails.length === 0) this.redirectToNewAccount()
        break
      }
    }
  }

  renderItem({ item }) {
    return (
      <Button
        icon={
          <Icon
            name='trash'
            size={20}
            style={{
              marginLeft: 20
            }}
            light
          />
        }
        iconRight
        buttonStyle={{ backgroundColor: 'transparent' }}
        titleStyle={{ color: 'grey', fontFamily: 'roboto', fontSize: 18 }}
        title={item}
        onPress={() => { this.deleteAccount(item) }} />
    )
  }

  componentDidMount() {
    this.retrieveEmails()
  }

  render() {
    return (
      <View style={styles.parentComponent}>
        <Grid style={{ height: hp('100%'), backgroundColor: '#fff' }}>
          <Col size={10} />
          <Col size={80} >
            <Row size={0.5} style={{ justifyContent: 'flex-start', alignItems: 'center', flexDirection: 'column' }}>
              <Text style={styles.welcomeText}>
                login.welcome1
              </Text>
              <Text style={styles.subWelcomeText}>
                login.welcome2
              </Text>
            </Row>
            <Row style={{ flexDirection: 'column', justifyContent: 'center' }}>
              <Text style={styles.deleteAccountText}>
                login.deleteAccount
              </Text>
              <FlatList
                contentContainerStyle={{ alignItems: 'flex-start' }}
                data={this.state.emails}
                renderItem={this.renderItem}
                keyExtractor={item => item}
                style={{ width: wp('75%') }}
              />
            </Row>
          </Col>
          <Col size={10} />
        </Grid>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  wrappingView: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  welcomeText: {
    fontSize: 24,
    color: '#444444',
    flexWrap: 'wrap',
    textAlign: 'center',
    fontFamily: 'Montserrat-Bold'
  },
  subWelcomeText: {
    fontSize: 15,
    color: '#444444',
    flexWrap: 'wrap',
    textAlign: 'center',
    fontFamily: 'Montserrat-Medium',
    width: '100%'
  },
  deleteAccountText: {
    marginBottom: 10,
    fontSize: 16,
    color: '#444444',
    flexWrap: 'wrap',
    fontFamily: 'roboto'
  },
  parentComponent: {
    backgroundColor: '#fff',
    flex: 1
  },
})