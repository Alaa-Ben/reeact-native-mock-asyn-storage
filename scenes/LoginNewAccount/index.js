import React, { Component } from 'react'
import {
  Text,
  View,
  ScrollView,
  Image,
  StyleSheet,
  FlatList,
  Platform,
  Linking,
  ActivityIndicator,
  TouchableHighlight
} from 'react-native'
import AsyncStorage from '@react-native-community/async-storage'

export default class LoginNewAccount extends Component {

  constructor(props) {
    super(props)
    this.state = {
      loginForm: {
        email: '',
        password: ''
      },
      loading: false,
      emails: [],
      isFocused: {
        email: false,
        password: false
      },
      disabled: true,
      credentials: null,
      rememberedEmail: null
    }
    this.redirectToDeleteAccount = this.redirectToDeleteAccount.bind(this)
    this.renderItem = this.renderItem.bind(this)
  }

  async retrieveEmails() {
    try {
      const values = await AsyncStorage.multiGet(['emails', 'credentials', 'rememberedEmail',
        'userCredentials', 'systemID', 'userID', 'domainName'])
      let emails_array, credentials, rememberedEmail, userCredentials, systemID, userID, domainName
      values.map((el) => {
        if (el[0] === 'emails') emails_array = el[1]
        if (el[0] === 'credentials') credentials = el[1]
        if (el[0] === 'rememberedEmail') rememberedEmail = el[1]
        if (el[0] === 'userCredentials') userCredentials = el[1]
        if (el[0] === 'systemID') systemID = parseInt(el[1])
        if (el[0] === 'userID') userID = parseInt(el[1])
        if (el[0] === 'domainName') domainName = el[1]
      })
      if (emails_array !== null) {
        this.setState({ emails: JSON.parse(emails_array), credentials, rememberedEmail },
          async () => {
                  AsyncStorage.multiRemove(['userCredentials', 'systemID', 'userID', 'domainName'])
          })
        }
    } catch (error) {
      console.log(error)
    }
  }

  storeEmail() {
    let emails = this.state.emails
    emails.push(this.state.loginForm.email)
    AsyncStorage.setItem('emails', JSON.stringify(emails))
  }

  redirectToDeleteAccount() {
    const { navigate } = this.props.navigation
    navigate('LoginDeleteAccount')
  }

  renderItem({ item }) {
    return (
      <TouchableHighlight
        onPress={() => { }} />
    )
  }

  componentDidMount() {
    this.retrieveEmails()
  }

  render() {
      return (
        <View style={styles.parentComponent} />
      )
  }
}

const styles = StyleSheet.create({
  parentComponent: {
    backgroundColor: '#fff',
    flex: 1
  }
})