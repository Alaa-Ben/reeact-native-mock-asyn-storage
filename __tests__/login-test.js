import 'react-native'
import React from 'react'
import LoginNewAccount from '../scenes/LoginNewAccount'
import LoginDeleteAccount from '../scenes/LoginDeleteAccount'
import MockedNavigator from '../__mocks__/MockedNavigator'

import { cleanup, render } from 'react-native-testing-library'

describe('Login', () => {
  afterEach(cleanup)
  it('LoginNewAccount renders correctly', () => {
    const { toJSON } = render(<MockedNavigator component={LoginNewAccount} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('LoginDeleteAccount renders correctly', () => {
    const { toJSON } = render(<MockedNavigator component={LoginDeleteAccount} />)
    expect(toJSON()).toMatchSnapshot()
  })
})