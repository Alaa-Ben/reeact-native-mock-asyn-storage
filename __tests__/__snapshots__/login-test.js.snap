// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login LoginDeleteAccount renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(224, 224, 224)",
                "flex": 1,
                "shadowColor": "rgb(224, 224, 224)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#fff",
                            "flex": 1,
                            "flexDirection": "row",
                            "height": 1334,
                          }
                        }
                      >
                        <View
                          size={10}
                          style={
                            Object {
                              "flex": 10,
                              "flexDirection": "column",
                            }
                          }
                        />
                        <View
                          size={80}
                          style={
                            Object {
                              "flex": 80,
                              "flexDirection": "column",
                            }
                          }
                        >
                          <View
                            size={0.5}
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 0.5,
                                "flexDirection": "column",
                                "justifyContent": "flex-start",
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#444444",
                                  "flexWrap": "wrap",
                                  "fontFamily": "Montserrat-Bold",
                                  "fontSize": 24,
                                  "textAlign": "center",
                                }
                              }
                            >
                              login.welcome1
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#444444",
                                  "flexWrap": "wrap",
                                  "fontFamily": "Montserrat-Medium",
                                  "fontSize": 15,
                                  "textAlign": "center",
                                  "width": "100%",
                                }
                              }
                            >
                              login.welcome2
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "flexDirection": "column",
                                "justifyContent": "center",
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#444444",
                                  "flexWrap": "wrap",
                                  "fontFamily": "roboto",
                                  "fontSize": 16,
                                  "marginBottom": 10,
                                }
                              }
                            >
                              login.deleteAccount
                            </Text>
                            <RCTScrollView
                              contentContainerStyle={
                                Object {
                                  "alignItems": "flex-start",
                                }
                              }
                              data={Array []}
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={false}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              stickyHeaderIndices={Array []}
                              style={
                                Object {
                                  "width": 562.5,
                                }
                              }
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View />
                            </RCTScrollView>
                          </View>
                        </View>
                        <View
                          size={10}
                          style={
                            Object {
                              "flex": 10,
                              "flexDirection": "column",
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Login LoginNewAccount renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(224, 224, 224)",
                "flex": 1,
                "shadowColor": "rgb(224, 224, 224)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "flex": 1,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
